src/ScheduledTasks/Models/ScheduleConfiguration.cs

```diff
@@ -0,0 +1,174 @@
+
+    /// <summary>
+    /// Gets or sets the instance ID of the orchestration function.
+    /// </summary>
+    public string? OrchestrationInstanceId { get; set; }
+
+    /// <summary>
+    /// Gets or sets the start time of the schedule.
+    /// </summary>
+    public DateTimeOffset? StartAt { get; set; }
+
+    /// <summary>
+    /// Gets or sets the end time of the schedule.
+    /// </summary>
+    public DateTimeOffset? EndAt { get; set; }
+
+    /// <summary>
+    /// Gets or sets the interval between schedule executions.
+    /// </summary>
+    public TimeSpan Interval
+    {
+        get => this.interval;
+        set
+        {
+            if (value <= TimeSpan.Zero)
+            {
+                throw new ArgumentException("Interval must be positive", nameof(value));
+            }
+
+            if (value.TotalSeconds < 1)
+            {
+                throw new ArgumentException("Interval must be at least 1 second", nameof(value));
+            }
+
+            this.interval = value;
+        }
+    }
+
+    /// <summary>
+    /// Gets or sets a value indicating whether to start the orchestration immediately when the current time is past the StartAt time.
+    /// By default it is false.
+    /// If false, the first run will be scheduled at the next interval based on the original start time.
+    /// If true, the first run will start immediately and subsequent runs will follow the regular interval.
+    /// </summary>
+    public bool StartImmediatelyIfLate { get; set; }
+
+    /// <summary>
+    /// Creates a new configuration from the provided creation options.
+    /// </summary>
+    /// <param name="createOptions">The options to create the configuration from.</param>
+    /// <returns>A new schedule configuration.</returns>
+    public static ScheduleConfiguration FromCreateOptions(ScheduleCreationOptions createOptions)
+    {
+        Check.NotNull(createOptions, nameof(createOptions));
+
+        return new ScheduleConfiguration(createOptions.ScheduleId, createOptions.OrchestrationName, createOptions.Interval)
+        {
+            OrchestrationInput = createOptions.OrchestrationInput,
+            OrchestrationInstanceId = createOptions.OrchestrationInstanceId,
+            StartAt = createOptions.StartAt,
+            EndAt = createOptions.EndAt,
+            StartImmediatelyIfLate = createOptions.StartImmediatelyIfLate,
+        };
+    }
+
+    /// <summary>
+    /// Updates this configuration with the provided update options.
+    /// </summary>
+    /// <param name="updateOptions">The options to update the configuration with.</param>
+    /// <returns>A set of field names that were updated.</returns>
+    public HashSet<string> Update(ScheduleUpdateOptions updateOptions)
+    {
+        Check.NotNull(updateOptions, nameof(updateOptions));
+        HashSet<string> updatedFields = new HashSet<string>();
+
+        if (!string.IsNullOrEmpty(updateOptions.OrchestrationName))
```

Comment:
should this be also checking if the value is different? Like
```csharp
if (!string.IsNullOrEmpty(updateOptions.OrchestrationName) 
    && updateOptions.OrchestrationName != this.OrchestrationName)
```

so that we can tell if the schedule is actually being updated.